src/avram/model.cr

```diff
@@ -99,6 +99,39 @@ abstract class Avram::Model
       :{{ type_declaration.var.stringify }}
     end
 
+    {% if type_declaration.type.stringify == "String" %}
+      {% value_generator = type_declaration.value %}
+
+      {% if !value_generator || value_generator && !(value_generator.is_a?(ProcLiteral) || value_generator.is_a?(ProcPointer) || value_generator.is_a?(Call)) %}
+          {% raise <<-ERROR
+              When using a String primary_key, you must also specify a way to generate the value.
+              You can provide a class method, a proc or a proc pointer.
+              Your value generator must return a non-nullable String.
+
+              Example:
+                #{@type.id} do
```

Comment:
I think this error would end up looking something like...

```
User do
  primary_key id : String
end
```

right? Maybe this just needs to be `table do` since it's just an example?

Reply:
Good catch